### ISSUE TITLE
Undeclared files

### ISSUE THREAD

**Issue by MichaelChambers** [fima-taf/vite-plugin-loading-script#1]
Apparently I have files that do not pass either of these tests.

`
		if (bundle[key].type === 'chunk') {
			//...
		} else if (bundle[key].type === 'asset' && bundle[key].fileName.endsWith('.css')) {
			//...
		}
		scriptCode +=	`parent.appendChild(${varName});`

`
The scriptCode includes parent.appendChild(file0) and (file1) without declaring them.

`
(function () {let scriptTag = document.getElementsByTagName('script');scriptTag = scriptTag[scriptTag.length - 1];const parent = scriptTag.parentNode;parent.appendChild(file0);parent.appendChild(file1);const file2 = document.createElement('script');file2.setAttribute('src', 'assets/index.0a9cd493.js');parent.appendChild(file2);const file3 = document.createElement('link');file3.setAttribute('href', 'assets/vendor.00fd496d.js');file3.setAttribute('rel', 'modulepreload');parent.appendChild(file3);const file4 = document.createElement('link');file4.setAttribute('href', 'assets/index.4a13278a.css');file4.setAttribute('rel', 'stylesheet');parent.appendChild(file4);})()
`

vite v2.5.5 building for production...
✓ 24 modules transformed.
dist/assets/index-legacy.9e292b9a.js       7.57 KiB / brotli: 2.49 KiB
dist/assets/polyfills-legacy.bca1f420.js   32.23 KiB / brotli: 11.92 KiB
dist/assets/vendor-legacy.c97384c0.js      48.62 KiB / brotli: 17.33 KiB
dist/assets/socks_blue.3653762e.jpg    141.55 KiB
dist/assets/socks_green.646eb412.jpg   136.80 KiB
dist/app.8dd2ffe4.js                   0.65 KiB
dist/index.html                        1.54 KiB
dist/assets/index.4a13278a.css         2.67 KiB / brotli: 0.69 KiB
dist/assets/index.0a9cd493.js          5.46 KiB / brotli: 2.07 KiB
dist/assets/vendor.00fd496d.js         48.87 KiB / brotli: 17.38 KiB

Perhaps it is due to the plugin-legacy files?
`
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import legacy from '@vitejs/plugin-legacy'
import { loadingScript } from 'vite-plugin-loading-script'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    legacy({
      targets: ['defaults', 'not IE 11']
    }),
    loadingScript(/** options **/)
  ]
})
`

**Comment by MichaelChambers** [comment]
It appears to be due to the 2 jpg files. I removed the legacy, and the new build output is below. The generated script still appends the 2 undeclared file0 and file1.

vite v2.5.5 building for production...
✓ 24 modules transformed.
dist/assets/socks_green.646eb412.jpg   136.80 KiB
dist/assets/socks_blue.3653762e.jpg    141.55 KiB
dist/app.8dd2ffe4.js                   0.65 KiB
dist/index.html                        0.48 KiB
dist/assets/index.4a13278a.css         2.67 KiB / brotli: 0.69 KiB
dist/assets/index.0a9cd493.js          5.39 KiB / brotli: 2.05 KiB
dist/assets/vendor.00fd496d.js         48.87 KiB / brotli: 17.38 KiB

**Comment by MichaelChambers** [comment]
As the jpgs obviously do not show up in the index.html head with the other script/link tags, they could be safely skipped, perhaps just add:
`else { continue }` 

I also noticed that index.html has:
`<script type="module" crossorigin src="/assets/index.0a9cd493.js"></script>`
Should that type="module" and crossorigin be added to the generated:
`const file2 = document.createElement('script');file2.setAttribute('src', 'assets/index.0a9cd493.js');parent.appendChild(file2);`

**Comment by fima-taf** [comment]
Hey @MichaelChambers
First of all thanks for using the plugin :)
The 'Undeclared files' issue got fixed (you were right with the `else { continue }` part missing - thanks).
About your last comment:
`type="module"` - I've add this attribute because rollup is ES6 based so it is pretty reasonable to add this attribute.
`crossorigin` - I think that there will be some people that won't need this attribute, so I've added it as an option in the config object.

The updated version is `1.0.2`
